### DISCUSSION TITLE
Version 3.9.0: Add short year formatter and fix year of day computation

### DISCUSSION THREAD

**Discussion by ChristopherRabotin** [nyx-space/hifitime#276] [Announcements]
## What's Changed
* Relax UNIX test + release 3.8.6 by @ChristopherRabotin in https://github.com/nyx-space/hifitime/pull/266
* Introduce %y formatter by @gwbres in https://github.com/nyx-space/hifitime/pull/268
* Update tabled requirement from 0.14.0 to 0.15.0 by @dependabot in https://github.com/nyx-space/hifitime/pull/269
* Update der requirement from 0.6.1 to 0.7.8 by @dependabot in https://github.com/nyx-space/hifitime/pull/252
* [Maybe breaking change] Fix day of year computation by @ChristopherRabotin in https://github.com/nyx-space/hifitime/pull/273
* docs(fix): fixed reversed description of duration magnitude by @ethanmsl in https://github.com/nyx-space/hifitime/pull/271

## New Contributors
* @ethanmsl made their first contribution in https://github.com/nyx-space/hifitime/pull/271

**Full Changelog**: https://github.com/nyx-space/hifitime/compare/3.8.6...3.9.0

<hr /><em>This discussion was created from the release <a href='https://github.com/nyx-space/hifitime/releases/tag/3.9.0'>Version 3.9.0: Add short year formatter and fix year of day computation</a>.</em>

**Comment by gwbres** [comment]
Awesome Chris 🚀 

side note: it takes me at least 20 mins to upgrade all my tools on new hifitime releases

##  Corrected Day Of Year (DOY)

RINEX filename conventions uses %yy or %YYYY, and {:03, %DOY} all the time.
The previous faulty DOY behavior did not help in supporting that. 
We can now easily generate standardized filenames:

![image](https://github.com/nyx-space/hifitime/assets/7270467/e7edb7bd-0176-48ad-8f9a-bb321ee8a355)
![image](https://github.com/nyx-space/hifitime/assets/7270467/b76bcd76-0191-48eb-aaa7-7ff3243614bc)


See the footnote for the remaining tweak I still have to do to achieve that.
   
 ## New %yy Epoch formatter introduce
 
 Will most likely simply some of our parsing procedure
 
 ## Next step ?
 
 The next step that would fix all the little tweaks we have to do, is to provide a method to decompose an Epoch in other than UTC timescale.
 If we stick to RINEX filename generation example: RINEX files are most of the time expressed in one GNSS time system like GPST. 
 When formatting this file, we need to retrieve %YY  %HH %MM of the first observation.

Doing this 
 ```rust
 let (yy, _, _, hh, mm, ss) = epoch.to_gregorian_utc();
```

returns all fields in UTC, while we want them in GPST in this example.
See rinex/src/epoch: fn epoch_decompose for the logic I need to implement as a work around

**Reply by ChristopherRabotin** [reply]
I'm glad this works.

For the next step, I think that's a good idea. I'll let you create an issue so we can keep track of it.

Now that there's the minor breaking change warning in the readme, I think that version 3.10 can integrate what we had initially planned for version 4.0.0, including changing the behavior of keeping the duration in the initialization time scale.
